ethereum/core/build.gradle

```diff
@@ -91,26 +91,32 @@ test {
   exclude 'tech/pegasys/pantheon/ethereum/core/TransactionTest.class'
 }
 
-def generateTestFiles(FileTree jsonPath, File resourcesPath, File templateFile, String pathstrip, String destination, String namePrefix) {
+def generateTestFiles(FileTree jsonPath, File templateFile, String pathstrip, String destination, String namePrefix) {
   def referenceTestTemplate = templateFile.text
 
   // This is how many json files to include in each test file
   def fileSets = jsonPath.getFiles().collate(5)
 
   fileSets.eachWithIndex { fileSet, idx ->
-    def resPath = resourcesPath.getPath().replaceAll("\\\\", "/")
-
     def paths = []
     fileSet.each { testJsonFile ->
       if (!testJsonFile.getName().toString().startsWith(".")) {
-        paths << testJsonFile.getPath().toString()
-          .replaceAll(resPath + "/", "")
+        String parentDirectory = testJsonFile.getParentFile().getName() + "/"
+        String testFile = testJsonFile.getName()
+        if(pathstrip == "BlockchainTests/"){
+          String generalStateTests = "GeneralStateTests/"
+          if(testJsonFile.getParentFile().getParentFile().getName().toString() == generalStateTests){
```

Comment:
This condition will never file.  The name of the parent file will never end in a path separator ("/").  You need to test for "GeneralStateTests" and then concat "GeneralStateTests/"

Reply:
nice catch. thanks. this was a result of my overzealous beautification, I've reimplemented your proposed changes and run it again on windows to ensure that it's got the right test coverage